lib/compress.js

```diff
@@ -7960,42 +7959,37 @@ merge(Compressor.prototype, {
                     }
                     if (!first.is_boolean(compressor) && !first.is_string(compressor)) break;
                 }
-                return make_node(AST_Array, self, {
-                    elements: self.args
-                });
+                return make_node(AST_Array, self, { elements: self.args });
               case "Object":
-                if (self.args.length == 0) {
-                    return make_node(AST_Object, self, {
-                        properties: []
-                    });
-                }
+                if (self.args.length == 0) return make_node(AST_Object, self, { properties: [] });
                 break;
               case "String":
-                if (self.args.length == 0) return make_node(AST_String, self, {
-                    value: ""
-                });
-                if (self.args.length <= 1) return make_node(AST_Binary, self, {
-                    left: self.args[0],
+                if (self.args.length == 0) return make_node(AST_String, self, { value: "" });
+                if (self.args.length == 1) return make_node(AST_Binary, self, {
                     operator: "+",
-                    right: make_node(AST_String, self, { value: "" })
+                    left: make_node(AST_String, self, { value: "" }),
+                    right: self.args[0],
                 }).optimize(compressor);
                 break;
               case "Number":
-                if (self.args.length == 0) return make_node(AST_Number, self, {
-                    value: 0
-                });
+                if (self.args.length == 0) return make_node(AST_Number, self, { value: 0 });
                 if (self.args.length == 1) return make_node(AST_UnaryPrefix, self, {
-                    expression: self.args[0],
-                    operator: "+"
+                    operator: "+",
+                    expression: make_node(AST_Binary, self, {
+                        operator: "+",
+                        left: make_node(AST_String, self, { value: "" }),
+                        right: self.args[0],
```

Comment:
clever workaround!

Reply:
Side note... comments would be useful for this and the other uncommented transforms

// Number(x) ---> +(""+x)

Reply:
Glad you like it 😉

Comments added in 773b7db869b6b6eae01c44d97866f96479051bc9

Reply:
I suppose the legacy `unsafe` `Number(x) ---> +x` transform could be used if you had an `ecma` option <= 5.

Reply:
Not a big deal &minus; it's still shorter, and I don't see this being a very common usage pattern.